json_serializable-3.5.2/lib/src/utils.dart

```diff
@@ -70,7 +71,7 @@ T enumValueForDartObject<T>(
   List<T> items,
   String Function(T) name,
 ) =>
-    items.singleWhere((v) => source.getField(name(v)) != null);
+    items.singleWhereOrNull((v) => source.getField(name(v)) != null);
```

Comment:
Why was this behavior changed? Previously it expected to always get a result, but now it doesn't, which I wouldn't expect to have changed with the new analyzer version.